testprojects/tests/python/pants/dummies/BUILD

```diff
@@ -7,3 +7,24 @@ python_tests(
   name = 'failing_target',
   sources = ['test_fail.py'],
 )
+
+python_library(
+  name = 'example_lib',
+  sources = 'example_source.py',
+)
+
+python_tests(
+  name = 'target_with_source_dep',
+  sources = ['test_with_source_dep.py'],
+  dependencies = [
+    ':example_lib',
+  ],
+)
+
+python_tests(
+  name = 'target_with_thirdparty_dep',
+  sources = ['test_with_thirdparty_dep.py'],
+  dependencies = [
+    '3rdparty/python:futures',
```

Comment:
Upsettingly, the problem here is that there are both a `futures` and a `future` library in Python, and the one you want is `future` :(

(After fixing that up, the test will still awkwardly fail in py2 because `builtins.str` is actually a custom type which _looks_ like a unicode but isn't... But you can do something like `assert isinstance(text_type("foo"), text_type)` which will pass :))

Reply:
Your knowledge of the `future` library makes me happy :)

Reply:
I'm a little alarmed that I hadn't clued myself into the `futures`/`future` split. I will say that I definitely assumed `str` wasn't going to be the same as `text_type`, as there would probably be a reason they were different, but at that point the test was failing because of the `futures` mistake so I just tossed it up last night because it was already failing for now-much-less-mysterious reasons.